File: programmers_level1/로또의_최고_순위와_최저_순위.cpp
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {

    int right{ 0 }, zero{ 0 };

    for (int num : lottos) {
        if (num == 0) ++zero;
        else {
            auto itr = find(win_nums.begin(), win_nums.end(), num);
            if (itr != win_nums.end()) ++right;
        }
    }

    vector<int> answer;
    answer.push_back(7 - (right + zero));
    answer.push_back(7 - right);

    for (int i=0; i<2; ++i) {
        if (answer[i] == 7) --answer[i];
    }
    return answer;
}

File: programmers/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    count = 0
    rank = {6:1, 5:2, 4:3, 3:4, 2:5, 1:6, 0:6}
    for i in range(len(lottos)):            
        if lottos[i] in win_nums:
            count += 1
    
    num_pos = lottos.count(0)
    
    answer.append(rank[count + num_pos])
    answer.append(rank[count])
    return answer
